src/library/scala/Enumeration.scala

```diff
@@ -279,6 +279,8 @@ abstract class Enumeration (initial: Int) extends Serializable {
       new ValueSet(nnIds.rangeImpl(from.map(_.id - bottomId), until.map(_.id - bottomId)))
 
     override def empty = ValueSet.empty
+    override def knownSize: Int = nnIds.size
```

Comment:
I'm assuming that `ValueSet` is small enough that `(nnIds: BitSet).size` is effectively constant time